jobbergate-api/jobbergate_api/apps/job_scripts/routers.py

```diff
@@ -272,6 +281,12 @@ async def job_script_delete(job_script_id: int = Query(..., description="id of t
     delete_query = job_scripts_table.delete().where(where_stmt)
     await database.execute(delete_query)
 
+    try:
+        del s3man_jobscripts[job_script_id]
+    except KeyError:
+        # There is no need to raise an error if we try to delete a file that does not exist
+        logger.warning(f"Tried to delete job_script={job_script_id}, but it was not found")
+
```

Comment:
No, because the delete request is issued for a row in the database. If that is successfully deleted, then the keyerror is incidental if the file was removed on s3 already.